pkg/config/controller_config.go

```diff
@@ -76,5 +83,57 @@ func (cfg *ControllerConfig) Validate() error {
 	if len(cfg.ClusterName) == 0 {
 		return errors.New("kubernetes cluster name must be specified")
 	}
+
+	if err := cfg.validateDefaultTagsCollisionWithTrackingTags(); err != nil {
+		return err
+	}
+	if err := cfg.validateExternalManagedTagsCollisionWithTrackingTags(); err != nil {
+		return err
+	}
+	if err := cfg.validateExternalManagedTagsCollisionWithDefaultTags(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cfg *ControllerConfig) validateDefaultTagsCollisionWithTrackingTags() error {
+	trackingTagKeys := sets.NewString(
+		"elbv2.k8s.aws/cluster",
+		"ingress.k8s.aws/stack",
+		"ingress.k8s.aws/resource",
+		"service.k8s.aws/stack",
+		"service.k8s.aws/resource",
+	)
+	for tagKey := range cfg.DefaultTags {
+		if trackingTagKeys.Has(tagKey) {
+			return errors.Errorf("tag key %v cannot be specified in %v flag", tagKey, flagDefaultTags)
+		}
+	}
+	return nil
+}
+
+func (cfg *ControllerConfig) validateExternalManagedTagsCollisionWithTrackingTags() error {
+	trackingTagKeys := sets.NewString(
```

Comment:
better to have a single source of truth for this.

Reply:
make sense, let me do the change